File: src/companies/forgotten/ConvertDoubleToRational.java
package companies.forgotten;

public class ConvertDoubleToRational {
    public void calculate(double number) {
        int multiplicator = 1;
        while (number * multiplicator != (int) (number * multiplicator)) {
            multiplicator *= 10;
        }

        // number/multiplicator
        int multiplicatedNumber = (int) (number * multiplicator);
        int largestCommonDivisor = 1;
        for (int i = 2; i <= multiplicatedNumber; i++) {
            if (multiplicatedNumber % i == 0 && multiplicator % i == 0) {
                largestCommonDivisor = i;
            }
        }

        int firstPart = multiplicatedNumber / largestCommonDivisor;
        int secondPart = multiplicator / largestCommonDivisor;

        System.out.println(firstPart + "/" + secondPart);

    }
}

File: src/companies/forgotten/Scala/ConvertDoubleToRational.scala
package companies.forgotten.Scala

class ConvertDoubleToRational {

  def count(number: Double): Unit = {
    var base = 1
    while (number * base != (number * base).toInt) {
      base *= 10
    }
    val numberToDivide = (number * base).toInt
    val largestCommonDiv = largestCommonDivisor(numberToDivide, base)
    println(numberToDivide/largestCommonDiv + " / " + base/largestCommonDiv)
  }

  private def largestCommonDivisor(number: Int, number2: Int):Int = {
    var largest = 1
    var i = 2
    while(i <= number){
      if(number % i == 0 && number2 % i == 0){
        largest = i
      }
      i += 1
    }
    largest
  }
}
